graylog2-server/src/test/java/org/graylog/testing/mongodb/MongoDBExtensionWithRegistrationAsStaticFieldTest.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.testing.mongodb;
+
+import com.mongodb.client.MongoDatabase;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.MethodOrderer.OrderAnnotation;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.mongodb.client.model.Filters.eq;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Testing programmatic extension registration with static fields has been moved to this separate file because
+ * we cannot use static fields on nested classes in {@link MongoDBExtensionTest}.
+ */
+// Required to make the instance ID check work
+@TestMethodOrder(OrderAnnotation.class)
```

Comment:
I think the test could be improved by moving the tests asserting that the same instance id is used to a nested class. Running tests in order is only relevant for them so `@TestMethodOrder` would only be needed on the nested class and we could also get rid of the `Order` annotation on the last two tests.
The static field for `instanceIds` would still have to go in the outer class, which is a bit ugly, but a trade off, I'd be fine with.

Reply:
:+1: